### ISSUE TITLE
forcing new election

### ISSUE THREAD

**Issue by jsloot** [acassen/keepalived#1089]
I have two raspberry pi nodes running with keepalived (MASTER and BACKUP).

`keepalived -v
Keepalived v1.3.2 (12/25,2016)

Copyright(C) 2001-2016 Alexandre Cassen, <acassen@gmail.com>

Build options:  PIPE2 LIBNL3 FRA_OIFNAME FRA_SUPPRESS_PREFIXLEN FRA_SUPPRESS_IFGROUP RTAX_QUICKACK IPV4_DEVCONF LIBIPTC LIBIPSET LVS LIBIPVS_NETLINK IPVS_DEST_ATTR_ADDR_FAMILY VRRP VRRP_AUTH VRRP_VMAC SOCK_NONBLOCK SOCK_CLOEXEC FIB_ROUTING INET6_ADDR_GEN_MODE SNMP_V3_FOR_V2 SNMP SNMP_KEEPALIVED SNMP_CHECKER SNMP_RFC SNMP_RFCV2 SNMP_RFCV3 SO_MARK`

Configuration of the MASTER-Node:
`vrrp_instance VI_3 {
  state MASTER
  #smtp_alert
  interface wlan0
  virtual_router_id 3
  priority 100
  virtual_ipaddress {
    10.1.1.112/24 brd 10.1.1.255 dev wlan0 label wlan0:0
  }
}
`

Configuration of the BACKUP-Node:
`vrrp_instance VI_3 {
  state BACKUP
  #smtp_alert
  interface wlan0
  virtual_router_id 3
  priority 50
  virtual_ipaddress {
    10.1.1.112/24 brd 10.1.1.255 dev wlan0 label wlan0:0
  }
}
`

In the logfile of the MASTER I can see:
`Dec 17 14:28:38 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election
Dec 17 14:28:38 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election
Dec 17 14:28:38 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election
Dec 17 14:29:14 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election
Dec 17 14:29:14 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election
Dec 17 14:30:26 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election
Dec 17 14:30:26 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election`

And in the logfile of the BACKUP-Node I can see:
`Dec 17 14:28:00 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Transition to MASTER STATE
Dec 17 14:28:00 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Received advert with higher priority 100, ours 50
Dec 17 14:28:00 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering BACKUP STATE
Dec 17 14:28:00 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:28:16 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.
Dec 17 14:28:36 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Transition to MASTER STATE
Dec 17 14:28:37 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering MASTER STATE
Dec 17 14:28:37 fhem2 avahi-daemon[220]: Registering new address record for 10.1.1.112 on wlan0.IPv4.
Dec 17 14:28:37 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:28:38 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Received advert with higher priority 100, ours 50
Dec 17 14:28:38 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering BACKUP STATE
Dec 17 14:28:38 fhem2 avahi-daemon[220]: Withdrawing address record for 10.1.1.112 on wlan0.
Dec 17 14:28:38 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:28:38 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.
Dec 17 14:28:38 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.
Dec 17 14:29:12 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Transition to MASTER STATE
Dec 17 14:29:13 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering MASTER STATE
Dec 17 14:29:13 fhem2 avahi-daemon[220]: Registering new address record for 10.1.1.112 on wlan0.IPv4.
Dec 17 14:29:13 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:29:14 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Received advert with higher priority 100, ours 50
Dec 17 14:29:14 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering BACKUP STATE
Dec 17 14:29:14 fhem2 avahi-daemon[220]: Withdrawing address record for 10.1.1.112 on wlan0.
Dec 17 14:29:14 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:29:14 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.
Dec 17 14:29:14 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.
Dec 17 14:30:24 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Transition to MASTER STATE
Dec 17 14:30:25 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering MASTER STATE
Dec 17 14:30:25 fhem2 avahi-daemon[220]: Registering new address record for 10.1.1.112 on wlan0.IPv4.
Dec 17 14:30:25 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:30:26 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Received advert with higher priority 100, ours 50
Dec 17 14:30:26 fhem2 Keepalived_vrrp[4173]: VRRP_Instance(VI_3) Entering BACKUP STATE
Dec 17 14:30:26 fhem2 avahi-daemon[220]: Withdrawing address record for 10.1.1.112 on wlan0.
Dec 17 14:30:26 fhem2 Keepalived_vrrp[4173]: Remote SMTP server [93.241.245.61]:25 connected.
Dec 17 14:30:26 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.
Dec 17 14:30:26 fhem2 Keepalived_vrrp[4173]: SMTP alert successfully sent.`

I flushed all iptables-rules with iptables -F.

Why the BACKUP-Node always switchs to MASTER and back?

BR Jörn

**Comment by pqarmitage** [comment]
It appears that the BACKUP-Node is intermittently not receiving adverts from the master node, and so it times out and transitions to master. I see that you are using wireless interfaces; are they re-associating with the access point? Is there anything in the system logs at the time the backup node transitions to master?

Your configuration shown above doesn't specify the advertisement interval, so it should be defaulting to 1 second, however the logs show
```
Dec 17 14:28:38 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election 
Dec 17 14:28:38 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election 
Dec 17 14:28:38 fhem Keepalived_vrrp[4093]: VRRP_Instance(VI_3) Received advert with lower priority 50, ours 100, forcing new election 
```
i.e. 3 adverts within 1 second. Are adverts being delayed, perhaps due to the wireless network re-syncing or resetting?

This does not appear to be a keepalived issue, but more an issue with your network.

**Comment by jsloot** [comment]
Hmm... I can't find "interval" in the vrrp_instance-section. Only in the vrrp_script-section.

**Comment by jsloot** [comment]
I think I could solve it with

advert_int 3
preempt_delay 5

**Comment by pqarmitage** [comment]
Yes, `advert_int` is the option I was meaning. If it solves your problem, or significantly reduces it, then please update this issue, and then I think it can be closed. However, you might want to look at what is happening on your wireless network, since things don't appear to be quite right.
